### PR TITLE
No filtering done in the queryApiLogByOrderReferenceAndType method

### PR DESCRIPTION
**Pull request by vhrychenko**
- Developer(s): @vhrychenko

- Ticket: https://spryker.atlassian.net/browse/ECO-2631

#### Please confirm

- [x] No new OS components - or they have been approved by the legal department

#### Documentation

- [x] Functional documentation provided or in progress?
- [x] Integration guide for projects provided or not needed?
- [x] Migration guides for all contained majors provided or not needed?

#### Release Table

   Module                | Release Type         | Constraint Updates         |
   :--------------------- | :------------------------ | :--------------------- |
   ArvatoRss  | minor                 |                       |

#### Release Notes

*Created ArvatoRss repository according to Spryker architecture rules.
*Fixed bug with no filtering 

-----------------------------------------

#### Module ArvatoRss

_**Patch:** Backwards-compatible bug fix_

Def of done (by the responsible developer):
- [x] All changes are backwards-compatible. An outdated code is marked as deprecated.
- [x] The change is isolated and not mixed with any cleanups or other changes.
- [x] New and changed facade methods covered by functional tests. / Has nothing to test.
- [x] New and changed complex business logic is covered by unit or integration tests. / Has nothing to test.

##### Change log

Improvements

- Created ArvatoRss repository according to Spryker architecture rules.
- Fixed bug with no filtering

### CODE REVIEW

**Review by gechetspr — APPROVED** [review]
Looks ok. But please check why Travis is red.

**Comment by dereuromark** [comment]
See https://travis-ci.org/spryker-eco/arvato-rss/jobs/511011006
Code sniffer, architecture sniffer etc, and even Spryker CI red.

File: src/SprykerEco/Zed/ArvatoRss/Communication/Plugin/Oms/Condition/IsStoreOrderSuccessfulPlugin.php [hunk 1]
```diff
@@ -43,36 +44,36 @@ public function check(SpySalesOrderItem $orderItem)
...
     }
 
     /**
      * @param string $orderReference
      *
-     * @return \Orm\Zed\ArvatoRss\Persistence\SpyArvatoRssApiCallLog
+     * @return \Generated\Shared\Transfer\ArvatoRssApiCallLogTransfer
      */
-    protected function getApiCallLogEntry($orderReference)
+    protected function getApiCallLogEntry(string $orderReference): ArvatoRssApiCallLogTransfer
```
Comment (lockw1n) [inline src/SprykerEco/Zed/ArvatoRss/Communication/Plugin/Oms/Condition/IsStoreOrderSuccessfulPlugin.php]: We do not prefer business logic in plugins.

File: src/SprykerEco/Zed/ArvatoRss/Persistence/ArvatoRssPersistenceFactory.php [hunk 2]
```diff
@@ -8,20 +8,28 @@
...
+    public function createArvatoRssApiCallLogQuery(): SpyArvatoRssApiCallLogQuery
     {
-        return SpyArvatoRssApiCallLogQuery::create()->orderByUpdatedAt(Criteria::DESC);
+        return SpyArvatoRssApiCallLogQuery::create();
+    }
+
+    /**
+     * @return \SprykerEco\Zed\ArvatoRss\Persistence\Mapper\ArvatoRssPersistenceMapperInterface
+     */
+    public function createArvatoRssPersistenceMapper(): ArvatoRssPersistenceMapperInterface
```
Comment (lockw1n) [inline src/SprykerEco/Zed/ArvatoRss/Persistence/ArvatoRssPersistenceFactory.php:33]: Persistence mappers do not require interfaces.

File: src/SprykerEco/Zed/ArvatoRss/Persistence/ArvatoRssRepository.php [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+     */
+    public function findApiLogByOrderReferenceAndType(string $orderReference, string $type): ?ArvatoRssApiCallLogTransfer
+    {
+        $arvatoRssApiCallLog = $this->getFactory()->createArvatoRssApiCallLogQuery()
+            ->filterByCallType($type)
+            ->filterByOrderReference($orderReference)
+            ->findOne();
+
+        if ($arvatoRssApiCallLog === null) {
+            return null;
```
Comment (lockw1n) [inline src/SprykerEco/Zed/ArvatoRss/Persistence/ArvatoRssRepository.php:31]: Would be nice return empty transfer instead.

Comment (vhrychenko) [reply src/SprykerEco/Zed/ArvatoRss/Persistence/ArvatoRssRepository.php:31]: Changed

File: src/SprykerEco/Zed/ArvatoRss/Persistence/ArvatoRssPersistenceFactory.php (hunk 2 above)
Comment (vhrychenko) [reply src/SprykerEco/Zed/ArvatoRss/Persistence/ArvatoRssPersistenceFactory.php:33]: Where I can found info about it?

File: src/SprykerEco/Zed/ArvatoRss/Communication/Plugin/Oms/Condition/IsStoreOrderSuccessfulPlugin.php (hunk 1 above)
Comment (vhrychenko) [reply src/SprykerEco/Zed/ArvatoRss/Communication/Plugin/Oms/Condition/IsStoreOrderSuccessfulPlugin.php]: I agree but I'm not sure we have time for creating business model right now.

File: src/SprykerEco/Zed/ArvatoRss/Persistence/ArvatoRssRepository.php (hunk 3 above)
Comment (gechetspr) [reply src/SprykerEco/Zed/ArvatoRss/Persistence/ArvatoRssRepository.php:31]: But this is a common practice in core to return null if we didn't find a record.

Comment (dereuromark) [reply src/SprykerEco/Zed/ArvatoRss/Persistence/ArvatoRssRepository.php:31]: I agree, an empty transfer cloaks the case that this is actually without data. So usually nullable is the cleaner approach isnt it?

File: src/SprykerEco/Zed/ArvatoRss/Persistence/ArvatoRssPersistenceFactory.php (hunk 2 above)
Comment (lockw1n) [reply src/SprykerEco/Zed/ArvatoRss/Persistence/ArvatoRssPersistenceFactory.php:33]: That was @PhilinTv recommendations/suggestions.

File: src/SprykerEco/Zed/ArvatoRss/Persistence/ArvatoRssRepository.php (hunk 3 above)
Comment (lockw1n) [reply src/SprykerEco/Zed/ArvatoRss/Persistence/ArvatoRssRepository.php:31]: Okay, core rules are changing rapidly. Looks like my knowledge is outdated.

Comment (vhrychenko) [reply src/SprykerEco/Zed/ArvatoRss/Persistence/ArvatoRssRepository.php:31]: Reverted

**Review by alex-galych — APPROVED** [review]

**Pull request merged**